contracts/src/types/RequestLock.sol

```diff
@@ -65,24 +58,24 @@ library RequestLockLibrary {
 
     function setProverPaidBeforeLockDeadline(RequestLock storage requestLock) internal {
         requestLock.requestLockFlags = PROVER_PAID_DURING_LOCK_FLAG;
-        // Zero out second slot for gas refund.
-        requestLock.price = uint96(0);
-        requestLock.stake = uint96(0);
-        requestLock.fingerprint = bytes8(0);
+        // Zero out slots 1-2 for gas refund.
+        clearSlot1And2(requestLock);
     }
 
     function setProverPaidAfterLockDeadline(RequestLock storage requestLock, address prover) internal {
         requestLock.prover = prover;
         requestLock.requestLockFlags = PROVER_PAID_AFTER_LOCK_FLAG;
-        // We don't zero out the second slot as stake information is required for slashing.
+        // We don't zero out slot 1 as stake information is required for slashing.
+        // Zero out slot 2 for gas refund.
+        // Clearing via requestLock.requestDigest = bytes32(0)
+        // would require touching slot 1 as well.
+        clearSlot2(requestLock);
```

Comment:
Since slashing will clear both slots anyway, why not leave this until slashing occurs?

> Clearing via requestLock.requestDigest = bytes32(0) would require touching slot 1 as well.

Is this true? I think the Solidity layout rules mean that, because `requestDigest` does not fit in slot1 entirely, it is aligned into slot2. Assuming I'm not mistaken`requestLock.requestDigest = bytes32(0)` should only touch slot2.

Reply:
> Since slashing will clear both slots anyway, why not leave this until slashing occurs?

I guess this is a question of who should receive the gas refund for clearing slot 2 in the "lock => lock expire => fulfill after lock => slash" case: the slasher or the fulfiller?

There is also the lock => lock expire => no fulfill case, which clearing both slots on slash was meant to cover.

I'm inclined to favor the fulfiller over the slasher

Reply:
> Is this true? I think the Solidity layout rules mean that, because requestDigest does not fit in slot1 entirely, it is aligned into slot2. Assuming I'm not mistakenrequestLock.requestDigest = bytes32(0) should only touch slot2.

Ah yes you are right. This approach is still cheaper, so I'm inclined to keep it, as its probably something we'd end up doing during mainnet gas golfing anyway